src/docfx/lib/Levenshtein.cs

```diff
@@ -1,26 +1,53 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
+// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
 using System.Buffers;
+using System.Collections.Generic;
 
 namespace Microsoft.Docs.Build
 {
     internal static class Levenshtein
     {
         /// <summary>
-        /// Calculate the Levenshtein Distance from src to target.
+        /// Calculate the Levenshtein Distance from source string to target string.
         /// </summary>
         /// <param name="src"> The source string </param>
         /// <param name="target">The target string </param>
         /// <returns>Levenshtein Distance</returns>
         public static int GetLevenshteinDistance(string src, string target)
+        {
+            int srcLength = string.IsNullOrEmpty(src) ? 0 : src.Length;
+            int targetLength = string.IsNullOrEmpty(target) ? 0 : target.Length;
+
+            if (srcLength == 0)
+            {
+                return targetLength;
+            }
+            if (targetLength == 0)
+            {
+                return srcLength;
+            }
+            var srcCharArray = src.ToCharArray();
+            var targetCharArray = target.ToCharArray();
+            return GetLevenshteinDistance<char>(srcCharArray, targetCharArray, Comparer<char>.Default);
+        }
+
+        /// <summary>
+        /// Get levenshtein distance of two generic type list.
+        /// </summary>
+        /// <typeparam name="T"> Generic type, e.g. string</typeparam>
+        /// <param name="src">The source type list</param>
+        /// <param name="target">The target type list</param>
+        /// <param name="comparer">Implementation of IEqualityComparer for given type</param>
+        /// <returns>Levenshtein distance of the two given list</returns>
+        public static int GetLevenshteinDistance<T>(IList<T> src, IList<T> target, IComparer<T> comparer)
```

Comment:
when is T not a string?

Reply:
maybe a list of integers? but there's no such use case yet.